src/System.IO.FileSystem.DriveInfo/src/System/IO/DriveInfo.Windows.cs

```diff
@@ -133,24 +121,15 @@ public static DriveInfo[] GetDrives()
         }
 
         // Null is a valid volume label.
-        public string VolumeLabel
+        public unsafe string VolumeLabel
         {
             get
             {
-                // NTFS uses a limit of 32 characters for the volume label,
-                // as of Windows Server 2003.
-                const int volNameLen = 50;
-                StringBuilder volumeName = new StringBuilder(volNameLen);
-                const int fileSystemNameLen = 50;
-                StringBuilder fileSystemName = new StringBuilder(fileSystemNameLen);
-                int serialNumber, maxFileNameLen, fileSystemFlags;
+                char* volumeName = stackalloc char[Interop.Kernel32.MAX_PATH + 1];
 
-                uint oldMode;
-                bool success = Interop.Kernel32.SetThreadErrorMode(Interop.Kernel32.SEM_FAILCRITICALERRORS, out oldMode);
-                try
+                using (DisableMediaInsertionPrompt.Create())
                 {
-                    bool r = Interop.Kernel32.GetVolumeInformation(Name, volumeName, volNameLen, out serialNumber, out maxFileNameLen, out fileSystemFlags, fileSystemName, fileSystemNameLen);
-                    if (!r)
+                    if (!Interop.Kernel32.GetVolumeInformation(Name, volumeName, Interop.Kernel32.MAX_PATH + 1, null, null, out int fileSystemFlags, null, 0))
                     {
                         int errorCode = Marshal.GetLastWin32Error();
                         // Win9x appears to return ERROR_INVALID_DATA when a
```

Comment:
Delete this one-off Win9x specific comment & code while you are on it? The other callsite does not have it.

Reply:
done